### ISSUE TITLE
Multiple Search Box accessibility issues

### ISSUE THREAD

**Issue by claycompton** [NuGet/NuGetGallery#1483]
# Keyboard and mouse experience on receiving focus are inconsistent in new search box

To repro, follow these steps.
    1. Type text in the textbox.
    2. Click outside the box to make it lose focus.
    3. Click inside the box to restore focus.  

RESULT:  The box won't immediately expand.  It will do so only if you type additional content.
EXPECTED: if you use the keyboard to tab or shift-tab back to the box, it will expand immediately on receiving focus.  We should be consistent between these two modalities.
# Accessibility (WCAG level A): Change of tab order when tabbing into and out of new search box.

To reproduce this, follow these steps:
1.  Navigate to http://qa.nugettest.org/.
2.  Using only the keyboard, tab to the search box.  Notice that each of the links in the navigation bar are tab stops.
3.  While focus is on the search box, press the space bar (to force the search box to expand)
4.  Press Shift-Tab to leave the search box.

RESULT:  Focus will move to one of the account buttons (either Register or Log Off), not the buttons in the navigation bar.
EXPECTED:  We shouldn't have unexpected changes in focus order, which will confuse blind users.  In this case, Shift-Tab ought to move focus to the Blog link in the navigation bar.
